src/porepy/grids/mdg_generation.py

```diff
@@ -0,0 +1,295 @@
+    utils.set_mesh_sizes(lower_level_args)
+
+    return (lower_level_args, extra_args_list, kwargs)
+
+
+def _preprocess_cartesian_args(omega, mesh_arguments, kwargs):
+
+    nx_cells = None
+    phys_dims = None
+
+    xmin = omega.bounding_box["xmin"]
+    xmax = omega.bounding_box["xmax"]
+    ymin = omega.bounding_box["ymin"]
+    ymax = omega.bounding_box["ymax"]
+
+    h_size = mesh_arguments["mesh_size"]
+    n_x = round((xmax - xmin) / h_size)
+    n_y = round((ymax - ymin) / h_size)
+
+    nx_cells = [n_x, n_y]
+    phys_dims = [ymax, xmax]
+
+    if omega.dim == 3:
+        zmin = omega.bounding_box["zmin"]
+        zmax = omega.bounding_box["zmax"]
+        n_z = round((zmax - zmin) / h_size)
+        nx_cells = [n_x, n_y, n_z]
+        phys_dims = [ymax, xmax, zmax]
+
+    # Avoid duplicity in values for keyword argument
+    nx_arg = kwargs.get("nx", None)
+    if isinstance(nx_arg, nx_cells.__class__) and len(nx_arg) == omega.dim:
+        # overwrite information if user provide a valid nx
+        nx_cells = nx_arg
+        kwargs.pop("nx")
+
+    physdims_arg = kwargs.get("physdims", None)
+    if isinstance(physdims_arg, phys_dims.__class__) and len(phys_dims) == omega.dim:
+        # overwrite information if user provide a valid physdims
+        phys_dims = physdims_arg
+        kwargs.pop("physdims")
+
+    # remove duplicate keys
+    [kwargs.pop(item[0]) for item in mesh_arguments.items() if item[0] in kwargs]
+
+    return (nx_cells, phys_dims, kwargs)
+
+
+def _preprocess_tensor_grid_args(omega, mesh_arguments, kwargs):
+
+    x_space = None
+    y_space = None
+    z_space = None
+
+    xmin = omega.bounding_box["xmin"]
+    xmax = omega.bounding_box["xmax"]
+    ymin = omega.bounding_box["ymin"]
+    ymax = omega.bounding_box["ymax"]
+
+    h_size = mesh_arguments["mesh_size"]
+    n_x = round((xmax - xmin) / h_size) + 1
+    n_y = round((ymax - ymin) / h_size) + 1
+
+    x_space = np.linspace(xmin, xmax, num=n_x)
+    y_space = np.linspace(ymin, ymax, num=n_y)
+
+    # Avoid duplicity in kwargs
+    x_arg = kwargs.get("x", None)
+    if isinstance(x_arg, x_space.__class__):
+        x_space = x_arg
+        kwargs.pop("x")
+    y_arg = kwargs.get("y", None)
+    if isinstance(y_arg, y_space.__class__):
+        y_space = y_arg
+        kwargs.pop("y")
+
+    if omega.dim == 3:
+        zmin = omega.bounding_box["zmin"]
+        zmax = omega.bounding_box["zmax"]
+        n_z = round((zmax - zmin) / h_size) + 1
+        z_space = np.linspace(zmin, zmax, num=n_z)
+
+        z_arg = kwargs.get("z", None)
+        if isinstance(z_arg, z_space.__class__):
+            z_space = z_arg
+            kwargs.pop("z")
+
+    # remove duplicate keys
+    [kwargs.pop(item[0]) for item in mesh_arguments.items() if item[0] in kwargs]
+
+    return (x_space, y_space, z_space, kwargs)
+
+
+def create_mdg(
+    grid_type: Literal["simplex", "cartesian", "tensor_grid"],
+    mesh_arguments: dict,
```

Comment:
IMO, a dictionary here is not the best choice in terms of data type. In fact, I would avoid `mesh_arguments: dict` and simply use `mesh_size: pp.number`.

Depending on the type of grid, internally, `mesh_size` will have to be translated to `mesh_arguments` (for simplices) or `number_of_cells` (for Cartesian grids). If the user wants to set, say, different mesh sizes close to a fracture via `mesh_size_frac` or close to the boundary via `mesh_size_bound`, they should be passed as **kwargs.

The documentation should be crystal clear on what means to give a single `mesh_size` and if there are default values assumed, e.g., meshing in 3d in the case of simplices will require `mesh_size_min`, but we can of course set this as `mesh_size * 0.1` (or something similar). Also, it should be clear what are the possible optional arguments available for each type of grid. The description of such arguments might be outsourced to each different class/method, but at least knowing what are the arguments I would say is rather useful.

Reply:
I see two questions being discussed here:
1. Where to document usage and (advanced) options. I agree that we need a tutorial on this. My suggestion is to update meshing_of_fractures.ipynb.
2. What's the best parameter type for setting mesh size. I would have to do a more thorough review to weigh in heavily on this. Immediately, your comment seems reasonable, Jhabriel's. But I don't see an explicit explanation as to why you'd avoid a dictionary.

Reply:
Based on the outcome of the core developer meeting, some arguments were moved to 'meshing_args' and their admissibility is tested and documented.